doc: explain supported os (#270)
Substra/substra-documentation: docs/source/substrafl_doc/substrafl_overview.rst

Substrafl and Substra are compatible with Python versions 3.8, 3.9 and 3.10 on Windows, MacOS and Linux.

.. note::

    Full support on the following OS:
        - **Ubuntu 22.06**
        - **MacOS** (*Intel* or *M1* chip)

    Support on Windows:
        - ``subprocess`` mode is tested on **Windows Server 2022**.

>         - ``docker`` mode is not tested in the daily process. Please refer to :ref:`local_mode` to use it.

Reviewer: ```suggestion
        - ``docker`` mode is not tested for now. Please refer to :ref:`local_mode` to use it.
```

        - ``remote`` mode is not tested for now.

> 

Reviewer: I would add something in the end:
Despite Windows not being deeply tested, fell free to [reach out](https://docs.substra.org/en/latest/additional/community.html) if you encounter any issue.
